src/ui/download-repo-dialog.cpp

```diff
@@ -275,45 +275,47 @@ bool DownloadRepoDialog::validateInputs()
         }
     }
 
-    if (manual_merge_mode_) {
-        return validateInputsManualMergeMode();
+    if (!validateInputDirectory()) {
+        return false;
     }
 
-    sync_with_existing_ = false;
```

Comment:
You should not move the large blocks of code below - it's very difficult to identify the real changes. Do something like

```cpp
if (manual_merge_mode_) {
    return true;
}

// The original code would be unchanged
// ...
```